### ISSUE TITLE
Problem with French Canada local settings on 1.7.5.1

### ISSUE THREAD

**Issue by 8ternity** [PrestaShop/PrestaShop#16004]
**Steps To Reproduce the issue**

1. Go to BO => International => Localization page
2. Import the Canada Package
3. Go to your Profile employee
4. Change the languge of the employee to Français CA (French)
5. Go to BO => Catalog => Products => Edit a product
6. See error in the console

**Describe the bug**
From the begining of using vers 1.7.5.X, i have some main.bundle.js errors with French Quebec's locales. 

The error happen in BackEnd, and sometimes FrontEnd.

I also test in the past to change locales. Prestashop 1.7.5.1 default installer, configure French Canada for iso "qc" and language "qc". It's clear supposed to be "fr" and "ca" (see screenshots).

**To Reproduce**
1. Go to 'PS Back Office'
2. Click on 'Advanced Settings'
3. Scroll down to 'Logs'
4. See error in navigation

**Screenshots**
![2019-10-17_10h34_36](https://user-images.githubusercontent.com/16780421/67018840-cc687000-f0c9-11e9-89a5-5f6aadf7d05e.png)

![2019-10-17_10h36_13](https://user-images.githubusercontent.com/16780421/67018981-fcb00e80-f0c9-11e9-8cd5-f1b833d03ee5.png)

![2019-10-17_10h38_11](https://user-images.githubusercontent.com/16780421/67019218-4b5da880-f0ca-11e9-837b-07c16100f058.png)

**Additional information**
PrestaShop version: 1.7.7.0
PHP version: 7.2.23
MySQLa version: 5.5.64-MariaDB

**Comment by khouloudbelguith** [comment]
Hi @8ternity,

Could you please navigate to your BO => International => Translations => Add / Update a language section => select "Français CA (French)" & click on the button "Add / Update a language".
![image](https://user-images.githubusercontent.com/16067358/67021672-461b5080-f100-11e9-9854-cf8575277ef5.png)

Thanks to check & feedback.

**Comment by 8ternity** [comment]
@khouloudbelguith 

Prestashop 1.7.5.X does not have French (Canada) language. They fixed on version 1.7.6.X by adding French Canada. 

See the available Translations in Prestashop :
![2019-10-17_12h56_32](https://user-images.githubusercontent.com/16780421/67030596-ba90c800-f0dd-11e9-9ce6-5c87d80bdb04.png)

The migration from 1.7.5.1 to 1.7.6 is not available. It's not supported yet. They have problem in Kernel differences (reported by offcial support). I already try it in a demo store i've dupplicate.

**Comment by khouloudbelguith** [comment]
@8ternity, using the PrestaShop version 1.7.5.1., in the Back office => International => Localization => when you try to import Canada pack, the French CA is not added?
https://drive.google.com/file/d/1YmGdNfkE-LrzUW8vNrnm3o4ZiJyHO_Np/view

Thanks!

**Comment by 8ternity** [comment]
@khouloudbelguith According to the store, the Canada is already set. But i import again and get an error (see video)

https://www.screenpresso.com/=cK3gd

Error :
Notice: SessionHandler::gc(): ps_files_cleanup_dir: opendir(/var/lib/php/session) failed: Permission denied (13)

**Comment by khouloudbelguith** [comment]
@8ternity, this is an issue with PHP and sessions folder, you need to check the error with your host
Thanks to check with them & feedback

**Comment by 8ternity** [comment]
@khouloudbelguith problem submited to my hosting provider. Will update post upon status.

**Comment by khouloudbelguith** [comment]
@8ternity, thanks!
Waiting for your feedback.

**Comment by 8ternity** [comment]
@khouloudbelguith Resolved. But French Canada not available, but i was able to import "French" again and with writing authorsation it's resolved.

**Comment by khouloudbelguith** [comment]
@8ternity, thanks for your feedback.
So, now after importing the Canada Package, all seems ok => no exception displayed?
![image](https://user-images.githubusercontent.com/16067358/67471546-de10c100-f64f-11e9-8ae7-d0f3fb982aa9.png)

Thanks!

**Comment by 8ternity** [comment]
@khouloudbelguith It seems that the error is still happened in Products section while editing a product.

So it has resolve in Logs, but in Editing a Product, still present :

![2019-10-28_09h08_46](https://user-images.githubusercontent.com/16780421/67681026-e3794e80-f962-11e9-87e4-7c94f6ae618a.png)

**Comment by khouloudbelguith** [comment]
Hi @8ternity,

Thanks for your feedback.
I found an error with PS1.7.5.1 & PS1.7.6.1 in the BO => Catalog => Product details page
![image](https://user-images.githubusercontent.com/16067358/67694967-928a4a00-f9a4-11e9-9afe-34f606c08185.png)
```
qc.js:1 Uncaught ReferenceError: file_not_found is not defined
    at qc.js:1
```
This issue occurs when `Français CA (French)` is the language selected by the employee.
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by 8ternity** [comment]
Great, i will wait for the fix. Thanks.

**Comment by 8ternity** [comment]
@khouloudbelguith I installed the version 1.7.6.2 in my store with default language has French Canada, and tiny_mce try to load qc.js for the language.   French Canada must be in fr.js and fr-CA has the iso. [According to ISO country and localisation](https://en.wikipedia.org/wiki/Language_localisation), French Canada must be set to fr-CA.

**Comment by khouloudbelguith** [comment]
@8ternity, This issue is different, can you open a new ticket.

Thanks!

**Comment by 8ternity** [comment]
It's not just different, language is bad configured in all version up to 1.7.6.2. Jquery will call same files when you edit a product to be translated.

**Comment by khouloudbelguith** [comment]
@8ternity, yes, I have the same issue with PS1.7.6.2
![image](https://user-images.githubusercontent.com/16067358/70132509-af412e00-1684-11ea-9dd9-4533fce03a89.png)

Thanks!

**Comment by 8ternity** [comment]
@marionf @khouloudbelguith I was renamed into MySQL ps_lang has follow, and Product Page error has been fixed. 

````iso has fr```` and ```` language_code to fr-CA````

![Language Patch in DB](https://user-images.githubusercontent.com/16780421/74671375-067d7780-5179-11ea-9e46-2bf410a84a97.png)

But, now, when i edit translations in "Conception > Internationalisation" and save a French label, i get a locale error.

![Error on French Canada label Saving](https://user-images.githubusercontent.com/16780421/74671411-1dbc6500-5179-11ea-85c0-f56aac19f488.png)

IMPORTANT: I was modify the locales, iso and language code **because Google Search Console is rejecting my French URL because it wasn't recognized by their bot for Organic Search.** 

![2020-02-17_12h00_26](https://user-images.githubusercontent.com/16780421/74673419-244cdb80-517d-11ea-86ad-0c038354e565.png)

It must have a solution just to inject locales fron fr_CA to be fixed after that? 

Thank You.

**Comment by 8ternity** [comment]
@marionf @khouloudbelguith Since i start to build our store in 2019, this issue wasn't resolved. I made 3 sales in the last 8 months. Today, im still getting issues with Prestashop locales. Im struggling to update product editor for english version and locales are causing error to update translations on modules and also on product editor. I get no help from prestashop community support, on Github, on Gitter, i can't get help to get rid off. Since 2019, i try update on test store a migration to 1.7.6.X and still getting issue if i migrate with 1-click upgrade. Its doesn't resolved the lcoales error for French Canada. I also, get developement request quotation with a developer company who can migrate our store from version 1.7.5.1 to 1.7.6.5. Im struggling to know if you resolve the issue on version 1.7.6.5 or i migrate to Magento CMS 2.X. I do undestand it's an Open Source software, but i maybe need to wait upon resolution, but we invest more than 1500$ USD in modules in your addons store. Now, then, is locales fully supported on version 1.7.6.5 and can i migrate all data (orders, products, features, theme) to newer version?

**Comment by Progi1984** [comment]
@LouiseBonnard @Junebyun  It seems a localization Problem, and you don't have been pinged on it.

**Comment by LouiseBonnard** [comment]
Hi @8ternity, and thanks @Progi1984, it looks like the problem has been solved in 1.7.6.5 because I can't reproduce this issue, I have `qc` as the language ISO code and `fr-CA` as the language code and the locale. Unless I did not properly follow the steps...

In any case, you can check in the [TranslationFiles](https://github.com/PrestaShop/LocalizationFiles), the [all_languages](https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/all_languages.json) file, and [locales](https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/legacy-to-standard-locales.json) file. Let me know if I got it wrong and I'll try again.

**Comment by 8ternity** [comment]
@LouiseBonnard If you let the French Canada wrong iso code from the clean installation Yes. But Google and Bing Bots doesn't recognized the language for French due to wrong iso code. You need to go to language and change it has follow, and try again.

![Locales for Language after 100% clean install](https://user-images.githubusercontent.com/16780421/86017589-33c51680-b9f2-11ea-89af-241f0c9e8a8b.png)

Change the setting for the proper locales after your installation :

![Set is has follow](https://user-images.githubusercontent.com/16780421/86017479-109a6700-b9f2-11ea-9b96-1904081a27fb.png)

Try again. Go to translate e-mails or prestashop for French Canada.

Thanks.

**Comment by 8ternity** [comment]
@LouiseBonnard Regarding [localisation](https://github.com/PrestaShop/PrestaShop/blob/develop/app/Resources/legacy-to-standard-locales.json) fr-ca must be included in 'fr' has fr-CA with the french. It has no indicates in localisation that fr-ca also include French Canada.

    "fr": "fr-FR",

for 

    "fr": "fr-FR","fr-CA",

?

It's why the localisation aren't recognized?

thanks.

**Comment by LouiseBonnard** [comment]
@8ternity, localization stands for localization packs and they are based on countries.

Regarding your first issue, I think I can see where the error is coming from in the file that installs the language, cf. https://github.com/PrestaShop/PrestaShop/blob/develop/install-dev/langs/qc/language.xml. Indicated language code should be

	<language_code>fr-ca</language_code>

instead of

	<language_code>qc</language_code>

@PrestaShop/prestashop-core-developers, can you confirm?

**Comment by Progi1984** [comment]
@LouiseBonnard I think `qc` directory should be renamed in `fr-CA` which is the ISO code. What do you think, team @PrestaShop/prestashop-core-developers ?

The language code is `fr`
The country code is `ca`.

The locale is so `fr-CA`.

**Comment by 8ternity** [comment]
@Progi1984 I confirm too. This is the good and proper language and iso code. For now, Google Shopping and Google Search Console not recognized our page.

**Comment by LouiseBonnard** [comment]
@Progi1984, thanks for the feedback! But renaming the directory would be considered as a breaking change, wouldn't it?

**Comment by 8ternity** [comment]
@LouiseBonnard Seriously? You're the only one e-Commerce still  having this issue. Magento 2.0 is working good. Did you know how much i loosing in SEO referral + Google Shopping due to wrong language that Google is refusing all my products in French about that issue.

**Comment by 8ternity** [comment]
@LouiseBonnard @Progi1984 It has to be made!

**Comment by 8ternity** [comment]
@Progi1984 @LouiseBonnard any ETA on how long it can take to be available or fixed? Is that something can be implement in 1.7.5.1 version or only in a newer version?

**Comment by LouiseBonnard** [comment]
Hi @8ternity, if it does not introduce any breaking change (still checking with the dev team to have a clue), it will be part of a future version. We keep you posted!

**Comment by 8ternity** [comment]
@LouiseBonnard You need one more step to reproduce. 'qc' iso code are not existing. You need to change for 'fr' like mentionned in [here](https://github.com/PrestaShop/PrestaShop/issues/16004#issuecomment-651177784). Change ISO code 'qc' to 'fr' and try to translate a module translation.

**Comment by 8ternity** [comment]
@LouiseBonnard Did you reproduce the problem?

**Comment by 8ternity** [comment]
@LouiseBonnard Any updates on the last comment?

**Comment by 8ternity** [comment]
@LouiseBonnard @Progi1984 Im sorry to write again, but i have problem reported with French customers that some importants parts are not translated in French. Here in Canada, in Quebec, we have the French Office Law Government that are obligated to offer to Quebec's residents a website available in French. It's the law and we can have fine if we don't resolved it. Since i report the problem last year, i do need that you focus on resolving this issue and find a solution, or resolve it quickly. You must also understand that my job is at risk. I can't tell another year to the company owner they need to wait again over and over for having a resolution. 

Could you please take this issue in priority? Thank you. Stephen.

**Comment by 8ternity** [comment]
@LouiseBonnard Can we have an answer please? Could you please try to reproduce with the step explain in reply [here](https://github.com/PrestaShop/PrestaShop/issues/16004#issuecomment-663502280)

**Comment by ttoine** [comment]
Hello @8ternity if this is a critical issue for your business, you could have proposed a pull request, or, if your company devs can't do it, you could have hired someone to fix it in a pull request. There are freelancers that can solve this kind of issue.

Please note that PrestaShop is an open source project. Anyone is welcome to improve it. It's not just the job of the PrestaShop company team.

The developers team working for the PrestaShop company is focused on specific priorities, selected after interviews with agencies, merchants, and other people from the ecosystem. Then, their priorities are defined by the Product Management team, based on identified needs for the company, its business model, partners, etc.

Of course, people using the PrestaShop project to build solutions and products for their customers might have specific use cases that needs specific fixes of custom development. That's what open source projects are for.

That said, if you need help to get started as a contributor, please let me know.

**Comment by 8ternity** [comment]
@ttoine Thanks for update. 

Our company does not have a developper, like many businesses choose to build their own store for selling products. Normaly business have project manager and peoples who manage the store. Since is a bug caused by a bad configuration in the store itself and affter product, store translations, it's a critical issue. I've communicate details over and over (sorry if my english are not perfect) but normaly it was well explained with captures and to do. 

Regarding the Pull Request, i don't know how to do that, but it's not your team need to do that? Or how we does that?

We try to hire two developper to fix that issue, but they don't know well locales how it gonna affect everything in the store. I get 2 quotes and one of them never want to touch to that issue.

Reagarding the issue, it's not an issue related to customisation, or, development request. It's a bad programmation call which Prestashop Dev's has build French Quebec language has localisation and Internationalisation. See [Here](https://www.localeplanet.com/icu/fr-CA/index.html) or [Windows Language Code Identifier](https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-LCID/%5bMS-LCID%5d-190313.pdf)

I do appreciate that you answer and given me some suggestions.

I hope that you understand that this issue need to be fixed by dev's, not reresenting to be customisation.

**Comment by ttoine** [comment]
> I hope that you understand that this issue need to be fixed by dev's, not reresenting to be customisation.

Well, that's the point. The problem, here, is that this issue is critical for you. However, the PrestaShop devs team will handle in priority issues that are critical for a loooooot of end users (merchants) or developers (modules developers, themes developers, ...).

That's why in this kind of case, the pull request to fix this issue should be written by the people with the need: you, or someone you hire. That's basically how open source projects are working: serious companies invest in the open source projects they use to build solutions and products to grow their business, because it's strategic for them. That means, among others, hiring developers to fix issues and contributing new features. Keep in mind: because most of the time, downloading open source software is free of charge, people think that the cost of ownership is zero. It's not true, and certainly the biggest misunderstandig.

The PrestaShop company is currently the biggest contributor to the PrestaShop open source project. But, it's not the role of the PrestaShop company to fix everything, free of charge: it would not be sustainable. We fix what is important for most users, we manage the road map for the sustainability of the project with a long term insight, and our developers spend a lot of time on review of the code. We are also managing the promotion of the project (events, marketing, ...). It's impossible to do everything. That's why many merchants (like you), agencies, freelancers, who built their business with the PrestaShop software are also contributing to it in many ways, including fixing issues.

Last thing about understanding open source: read the license. When you are using the PrestaShop software (note that it's the same for any other open source software), you, the user, are responsible of the use you have of the software. The PrestaShop company (or the editors of other open source software) is not accountable for issues you find. That's also why companies who use open source softwares are working with agencies or freelancers, or hire devs: to have a good technical support.

If you are a too small company to have a full time dev, go with freelancers. And, if you struggle to find someone who has the skill and experience to fix such an issue, then let me now. Maybe one of our partner can help you with that.

**Comment by 8ternity** [comment]
@ttoine So what you tell me is you will not resolve this issue due to priority? Never?

**Comment by 8ternity** [comment]
@PierreRambaud @ttoine  Can i know if you plan to resolve the issue or you left the problem like this? It's just like i explains it's not related to customisation, it's bad settings used in your localisation, not really related to customization.

**Comment by PierreRambaud** [comment]
@8ternity Unfortunately, we don't plan to resolve this issue right now. 
There are a lot of issues due to bad coding, legacy, wrong choices, etc. And this one is not our priority, because even if it concerns your business, we must fix more important problems before. When I use the terms "more import problems", it's not that yours is not important, it's only regarding what the agencies have told us.

As @ttoine said, it's an open-source project and everyone can contribute, if you can hire a developer to fix this issue, we will be glad to review and merge it as fast as possible.

**Comment by 8ternity** [comment]
This is exactly what i've deduce from all your answers! From that situation, i will make a YouTube video about the situation. Thanks for answering and confirming it.

**Comment by ttoine** [comment]
@8ternity do you use some module to advertise?

**Comment by PierreRambaud** [comment]
> This is exactly what i've deduce from all your answers! From that situation, i will make a YouTube video about the situation. Thanks for answering and confirming it.
> 
> Also, i'm receiving email from Prestashop with "How to advertise on Google Shopping and Facebook via PrestaShop Ads?" But it's not possible in French Canada because wrong locales. Google Shopping has refused all my pages due to wrong locales (which i submit in this issue [Here](https://user-images.githubusercontent.com/16780421/74673419-244cdb80-517d-11ea-86ad-0c038354e565.png).
> 
> ITs look like you don't know how this issue is affecting a lot of things, just no translations. Google has answered me that i do need to have the proper iso code for getting Google Shopping.


Feel free to create a YouTube video about the situation if it can help you to find someone to fix this bug.

I will quote @ttoine:

_The PrestaShop software is a project anyone can download to use as a base. It is not at all a ready to use the product with support and SLA, not at all a SaaS, not at all a freeware. So, you are actually using the project to build a solution to buy/sell online. And by the license, you are responsible for the use you have of the software._

_Well, again, if this is a serious problem for your business, you can fix it (the code is open and the license allows you to modify it) and, eventually be fair with the project and share the fix in a contribution to the project (the license is made to manage collaborative writing of the software)_

_There are currently around 300000 online shops running PrestaShop globally. Small, medium, but also very big companies. So it means that accountants and merchants have found a way to work with it unless it's fixed. That's why using open source software is interesting._


Please, just keep in mind this is an **open-source** project, as far as I know you don't pay me, or anyone here to do a job :sweat_smile:. 
This is why, if you need something specific, or want that old bug to be fixed as soon as possible, you must do it yourself, or if you're not able to do it, pay someone for this task.

**Comment by 8ternity** [comment]
@PierreRambaud Hi,

I do understand the concept of Open Source and by providing the software as-is, but in reality, it's the base settings that having the issue. It's normal to ask for main developper to submit and ask for a fix. Again, like i'm living an issue, we're French Canadian. Under your 300,000 online shop, you have not a lof of customer who used French Canadian. 

About the contribution, we spent over 3K in modules and Prestashop gain commissions on every modules purchased. So regarding working for free, some of you maybe work for pennies, but if i was PHP programmer, coding for helping peoples is for me great gratitude and great motivation. Samething for new features, newer technology. 

I went to a lot of support website without success. I went on Gitter 'Prestashop', Prestashop community forum, also 2 other official community support and nobody has the knowledge to fix this issue.

Also, we hire 2 company to investigate and how to fix that issue and their have no way to know how to resolve the issue because it's affecting to much programming. Many freelancers programmer have the knowledge to use API and modules guides, but working on Core of the programming, nobody seem's to know how to resolved it.

I'm pretty sure it's not a huge task. Maybe to add locales folder, point some javascript to correct locales for fr-CA. Samething for jquery or PHP. I'm a HTML and CSS programmer, but i don't have knowledge to resolve that.

Keep im mind that i already ask two specialist of Prestashop and Magento programmer, they also develop modules for me, but that issue is more main Core of the software and they have no clue how it can be fixed and what's it will affecting.

**Comment by prestaforum** [comment]
@8ternity Hi, there is something that I fail to understand here. Please allow me to explain.

PrestaShop is opensource software. This means that you paid nothing to get it. Zero dollars, zero euros, zero whatever.

This is a software that, from what I can see, receives regularly new features.
And you get these features for free. Zero dollars.

This is a software that, from what I can see, receives regularly [bug fixes](https://github.com/PrestaShop/PrestaShop/pull/20688).
And you get these bug fixes for free. Zero dollars.

This is a software that, from what I can see, receives regularly [security updates](https://build.prestashop.com/news/prestashop-1-7-6-8-maintenance-release/).
And you get these security patches for free. Zero dollars.

Did you ever consider how expensive it would be to do this all by yourself ? If you could not rely on an opensource e-commerce website, if you had to write all this code by yourself, and add new features by yourself and fix issues by yourself and check the security by yourself ? How much time and money you would actually spend ? Well, thanks to the fact that this is open source, you get all of this in zero seconds and for zero dollars.

So ... on what ground can you actually threaten those fine foks from PrestaShop ? What do you plan to put in this Youtube video, "How I try to give a bad image to the people who gave me so much for free" ? (I assume this Youtube video you speak about aims to give bad image of prestashop)

What is the next step ? Will you find an issue in PHP and, if PHP team does not fix it, you make a Youtube video about it ? Then will you find an issue in Linux and, if Linux team does not fix it, you make a Youtube video about it ?

Be careful, I heard that Linus Torvalds is not the kind of man you want to mess with.

If you're not happy with open source software, there are alternatives: nobody is forcing you to use it. You have multiple options: use one of the competitors (WooCommerce, Magento are the 1st I can think of) or fork this project and start your own better version!

But rather than doing that, you're complaining here, threatening people who, again, are giving you so much for free, because it does not work for you! What went so wrong that you believe this is the right attitude.

What you will likely achieve is simply kill whatever motivation these people have and kill the project. You think this will never happen ? Read about that:
https://nolanlawson.com/2017/03/05/what-it-feels-like-to-be-an-open-source-maintainer/
https://www.theregister.com/2020/01/21/rust_actix_web_framework_maintainer_quits/

In this world, open source is one of the best things we have. You have a website ? Great, it runs because of hundred of open source projects. Apache, mysql, PHP, C, all of this you can use it for free and some people update/fix/secure it for you. For zero dollars, once again.

I think you should rather be thanking them for all of the stuff you got for free rather than complaining about one single issue.

**Comment by 8ternity** [comment]
Close that case! Thanks for the brutal and rough communication.

Also, thanks for threatening me.

Close the case!

**Comment by SimonGrn** [comment]
Who threatened you and where ?

**Comment by 8ternity** [comment]
> Be careful, I heard that Linus Torvalds is not the kind of man you want to mess with.

Sorry for problems and please just close the case. Thanks for your time.

**Comment by eternoendless** [comment]
@8ternity, I think you misunderstood @prestaforum's message.

You may not know who Linus Torvalds is: he's the creator of Linux and he's known for being very intolerant and hard on people. @prestaforum used him to illustrate that compared to other very well known figures of the open source community, we're actually quite nice.

For what is worth, to the best of my knowledge, neither him nor @prestaforum are directly affiliated with the PrestaShop project nor the PrestaShop company.

The bottom line is: yes we understand there is a problem, and that it's important to some people. But we can do only so much and it's not currently in our top priority list, so like in every other open source project, you can either patiently wait for someone to fix it for free, or find the way to fix it on your own (eg. pay an expert).

**Comment by 8ternity** [comment]
@eternoendless Thanks for the answer.

I have nothing else to say.. Thanks. 

Note: i do have 2 issues submited (one related to web service which is under moving to another API programmation).

Has for me i will stop reply and close my issue and stop following them. We will move to another plateform.

**Comment by hibatallahAouadni** [comment]
Hello,

I managed to reproduce the issue with PS1.7.6.9, PS1.7.7.0-rc.1+build.4 and develop, see the attached screen record below:

https://drive.google.com/file/d/11OSBXUhE2Y2Svu0HgIO3K0Gsf3zhgBIb/view

@PrestaShop/product-team what we should choose as status for this issue? **`ToDo`** or **`TBS`**?

Thanks!

**Comment by 8ternity** [comment]
Just a reminder, locales code and language code need to changed like this : 
[Refer to here](https://github.com/PrestaShop/PrestaShop/issues/16004#issuecomment-651177784)

**Comment by 8ternity** [comment]
I tested all the latest version 1.7.81 and they all still have qc has iso country and language code.

![Prestashop 1.7.8.1](https://user-images.githubusercontent.com/16780421/143062156-f1e5c961-2c21-401c-b4ca-ef72855d8d39.png)

**Comment by 8ternity** [comment]
Today i checked the latest version 1.7.8.5 and still the same configuration.

![Prestashop 1.7.8.5](https://user-images.githubusercontent.com/16780421/163440202-80d4296e-44c3-446f-9739-068a9284a514.png)

**Comment by 8ternity** [comment]
HI, Does this issue is planned and will be resolved? Im waiting for upgrading my e-Commerce 1.7.5.1 to a newer version and re-purchase all modules but since its not fixed, i can't upgrade to a newer and fastest version.

**Comment by matks** [comment]
Hello @8ternity I am sorry, this is probably not the answer you're looking for but this bug is not in the scope of the maintainer team for a release soon as far as I know.

It does not mean it's not an important bug, it's just that we have other more important topics to deliver 😞 .

But you don't have to wait for the maintainer team to work on this. As this is an open source project, one upside is that [every item needed to perform the repair is available](https://youtu.be/a8fHgx9mE5U): the source code, the documentation, the tests. If you can find a PrestaShop skilled developer ready to repair this for you, he has access to everything he needs 👍 . But I think I remember discussing with you and you already tried this if my memory is correct.

**Comment by marsaldev** [comment]
@8ternity just checked on the v8 (dev branch) and the language code is `fr-ca`
Are the `iso-code` and `language code` correct (in the ss)?
<img width="917" alt="Schermata 2022-05-12 alle 11 14 48" src="https://user-images.githubusercontent.com/48283112/168036525-b85e6942-7fd0-4245-ad45-2160e27eb064.png">

**Comment by 8ternity** [comment]
Hi @marsaldev :)

Yes the **Language code** 'fr-ca' is correct. Only the iso-code need to fixed for 'fr'.


Thanks for your reply.

**Comment by Progi1984** [comment]
I think this bug is dependent to this one #16070.

**Comment by 8ternity** [comment]
Updates: I've been waiting more than 3 years for getting fixes and waiting for a proper Prestashop Version that will fixed our issues and we've been taken the decision to abandonned Prestashop e-Commerce CMS plateform. We invest tousands $ in modules and get custom developement for pricing modules, time and efforts and we've taken the decision to shut down our e-commerce this morning. I understand that it's a Open Source Community Project, but everytime we faced an issue, it was never resolved and we've been forced to use external software to add or edit products because some issues. We've hired 3 differents developper to try to fix this issue of Iso language, due to core construction, all developpers told us the samething, to wait for you to rework the CMS because the iso language and language code need to be fixed in a lot of placed, same for jQuery. At my opinion, it was the most easiest e-Commerce CMS to get multilanguages support to work with. Anyway so for now you can close this ticket and do whatever you want to do with it. Thanks.

We invest a lot in modules purchased, theme, file importer, web service module, payment solutions modules, carrier payment module, filter modules, Elastic Search Module, Google Merchant Module, etc.

![2022-11-01_10h12_37](https://user-images.githubusercontent.com/16780421/199254646-5a31ce77-c71e-4634-91d3-e42371178ab3.png)
![2022-11-01_10h16_35](https://user-images.githubusercontent.com/16780421/199254790-64ecc24e-e9ae-42d1-8e99-a9c7b6d52197.png)
![2022-11-01_10h17_19](https://user-images.githubusercontent.com/16780421/199254929-fbc1e9f4-0eec-47da-8ed1-faf33e9c3fb5.png)
